### PR TITLE
Hotfix/remove memory leak dual pad mode

### PR DESCRIPTION
**Pull request by rrcarlosrodriguez**

### CODE REVIEW

**Comment by rrcarlosrodriguez** [comment]
Intended to fix #318

File: gst-libs/gst/r2inference/gstvideoinference.c [hunk 1]
```diff
@@ -907,7 +907,7 @@ gst_video_inference_process_model (GstVideoInference * self, GstBuffer * buffer,
     GST_ELEMENT_ERROR (self, STREAM, FAILED,
         ("Subclass didn't implement post-process"), (NULL));
     ret = GST_FLOW_ERROR;
-    goto buffer_free;
+    goto out;
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstvideoinference.c:910]: We don't have valid `prediction_data` here.

File: gst-libs/gst/r2inference/gstvideoinference.c [hunk 2]
```diff
@@ -976,13 +976,12 @@ gst_video_inference_process_model (GstVideoInference * self, GstBuffer * buffer,
...
       priv->src_model);
 
-prediction_free:
-  g_free (prediction_data);
-
 buffer_free:
   gst_buffer_unref (buffer_model);
 
 out:
+  g_free (prediction_data);
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstvideoinference.c:983]: I'm unsure of breaking the single-input single-output symmetry. That is, resources are allocated in `A->B->C` order and deallocated in reverse order. `C->B->A`. Would it be better to keep the `prediction_free` label and have queue case save a reference and jump to `prediction_free`? Thoughts?

File: gst-libs/gst/r2inference/gstvideoinference.c (hunk 1 above)
Comment (rrcarlosrodriguez) [reply gst-libs/gst/r2inference/gstvideoinference.c:910]: It doesn't  matter, g_free will handle NULL pointer gracefully

File: gst-libs/gst/r2inference/gstvideoinference.c (hunk 2 above)
Comment (rrcarlosrodriguez) [reply gst-libs/gst/r2inference/gstvideoinference.c:983]: prediction_data will always need to be freed, I don't think there is a need to set an extra label and an extra jump, Might confuse for further extension. Looking for a bit of logic simplification here

**Pull request merged**